CorsixTH/Src/run_length_encoder.cpp

```diff
@@ -25,79 +25,79 @@ SOFTWARE.
 #include <new>
 #include <algorithm>
 
-IntegerRunLengthEncoder::IntegerRunLengthEncoder()
+integer_run_length_encoder::integer_run_length_encoder()
 {
-    m_pBuffer = nullptr;
-    m_pOutput = nullptr;
-    _clean();
+    buffer = nullptr;
+    output = nullptr;
+    clean();
 }
 
-IntegerRunLengthEncoder::~IntegerRunLengthEncoder()
+integer_run_length_encoder::~integer_run_length_encoder()
 {
-    _clean();
+    clean();
 }
 
-void IntegerRunLengthEncoder::_clean()
+void integer_run_length_encoder::clean()
 {
-    delete[] m_pBuffer;
-    delete[] m_pOutput;
-    m_pBuffer = nullptr;
-    m_pOutput = nullptr;
-    m_iRecordSize = 0;
-    m_iBufferSize = 0;
-    m_iBufferSizeUsed = 0;
-    m_iBufferOffset = 0;
-    m_iOutputSize = 0;
-    m_iOutputSizeUsed = 0;
-    m_iObjectSize = 0;
-    m_iObjectCount = 0;
+    delete[] buffer;
+    delete[] output;
+    buffer = nullptr;
+    output = nullptr;
+    record_size = 0;
+    buffer_capacity = 0;
+    buffer_size = 0;
+    buffer_offset = 0;
+    output_capacity = 0;
+    output_size = 0;
+    object_size = 0;
+    object_copies = 0;
 }
 
-bool IntegerRunLengthEncoder::initialise(size_t iRecordSize)
+bool integer_run_length_encoder::initialise(size_t iRecordSize)
 {
-    _clean();
-    m_iRecordSize = iRecordSize;
+    clean();
+    record_size = iRecordSize;
 
     // Buffer must hold at least 7 + 2 * 8 records, as the maximum object size
     // is 8 records, 2 of which are needed to detect a repeat, and 7 for the
     // offset at which the objects are found.
```

Comment:
so this sets requirements for `iRecordSize`. Imhi a better solution is to silently fix any given record size to the requirements.